index.d.ts

```diff
@@ -0,0 +1,166 @@
+import * as React from "react";
+
+interface ProviderProps {
+  apiKey: string;
+  children?: React.ReactNode;
+}
+
+export class StripeProvider extends React.Component<ProviderProps, {}> {}
+
+export function injectStripe<Props extends Object>(
+  WrappedComponent: React.ComponentClass<Props & StripeProps>,
+): React.ComponentClass<Props>;
+
+export interface StripeProps {
+  stripe: StripeShape;
+}
+
+type CreateTokenResult = {
+  token: Token;
+  error: void;
+} | {
+  // TODO: StripeError can be more specific.
+  error: any;
+  token: void;
+};
+
+export interface StripeShape {
+  createToken(options: CreateTokenOptions): Promise<CreateTokenResult>;
+}
+
+export class Elements extends React.Component<{}, {}> {}
+export class CardElement extends React.Component<CardElementOptions, {}> {}
+export class CardNumberElement extends React.Component<ElementOptions<void>, {}> {}
+export class CardExpiryElement extends React.Component<ElementOptions<void>, {}> {}
+export class CardCVCElement extends React.Component<ElementOptions<void>, {}> {}
+export class PostalCodeElement extends React.Component<PostalCodeElementOptions, {}> {}
+
+// stripe.js v3 type definitions below.
```

Comment:
I think the types below this point should be for stripe.js v3 (maybe in `@types/stripe`?), instead of in this file directly. (https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#packaging-dependent-declarations)